internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -111,9 +111,6 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 			pluginsdk.ForceNewIfChange("network_profile.0.network_plugin_mode", func(ctx context.Context, _, new, meta interface{}) bool {
 				return !strings.EqualFold(new.(string), string(managedclusters.NetworkPluginModeOverlay))
 			}),
-			pluginsdk.ForceNewIfChange("network_profile.0.network_policy", func(ctx context.Context, old, new, meta interface{}) bool {
-				return old.(string) != "" || new.(string) != string(managedclusters.NetworkPolicyCilium)
-			}),
```

Comment:
We can't remove this since this would inadvertently expose some preview functionality. We can extend this to allow updating from no network policy to azure or calico without recreating the cluster.